### ISSUE TITLE
Some overflow in pretrain model

### ISSUE THREAD

**Issue by kingsword5566** [mailong25/self-supervised-speech-recognition#20]
Hi Maillong,
I am in the stage of pretrain model (use config/pretraining/wav2vec2_base_librispeech.yaml)
Total duration of unlabeled data is ~100 hours but there are ~60% shorter than 5sec.
The only different setting of environments is that I fix the `distributed_world_size` from 64 to 1 (I only have one GPU). Others are same.
In the begin of training, there are some WARNING about overflow:
...
2021-01-26 03:01:31 | INFO | fairseq.trainer | begin training epoch 1
2021-01-26 03:02:32 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 64.0
2021-01-26 03:03:23 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 32.0
2021-01-26 03:04:15 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 16.0
2021-01-26 03:10:26 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 8.0
2021-01-26 03:56:38 | INFO | fairseq.trainer | NOTE: overflow detected, setting loss scale to: 4.
...

Two questions:
1. Is that means I should tune our `lr` because the gradient is out-of-range in FP16 or ignore this warning?
2. Our `train_ups` is really small `=0.02`. Is that normal?

**Comment by mailong25** [comment]
Hi there
1. this is normal , just ignore this warning. Because we fine-tune the model, the lr should be small. 
2. Not sure what you mean by train_ups. You should take a look at the loss value too see if it decrease significantly or not.

**Comment by kingsword5566** [comment]
Hi
Thanks for replying!
2.
`train_ups` represents updates per second in training.
After about 50 epochs, my `valid_loss` decrease from `3.358` to `3.172` , `train_loss` decrease from `3.549` to `3.387`
Is that normal? What stage should I stop the training?

**Comment by mailong25** [comment]
If your unlabeled data is in English, then it is normal becuase the init_model is already good.
